// @ionic/react/dist/components/index.js
import { addIcons } from 'ionicons';
import { ICON_PATHS } from 'ionicons/icons';
import { defineCustomElements } from '@ionic/core/loader';
import { createReactComponent } from './createComponent';
// createControllerComponent
export { default as IonAlert } from './IonAlert';
export { default as IonLoading } from './IonLoading';
export { default as IonToast } from './IonToast';
// createOverlayComponent
export { default as IonActionSheet } from './IonActionSheet';
export { default as IonModal } from './IonModal';
export { default as IonPopover } from './IonPopover';
// Custom Components
export { default as IonPage } from './IonPage';
export { default as IonTabs } from './navigation/IonTabs';
export { default as IonTabBar } from './navigation/IonTabBar';
export { IonRouterOutlet, IonBackButton } from './navigation/IonRouterOutlet';
addIcons(ICON_PATHS);
defineCustomElements(window);
// ionicons
export const IonIcon = createReactComponent('ion-icon');
// createReactComponent
export const IonTabBarInner = createReactComponent('ion-tab-bar');
export const IonRouterOutletInner = createReactComponent('ion-router-outlet');
export const IonBackButtonInner = createReactComponent('ion-back-button');
export const IonTab = createReactComponent('ion-tab');
export const IonTabButton = createReactComponent('ion-tab-button');
export const IonAnchor = createReactComponent('ion-anchor');
export const IonApp = createReactComponent('ion-app');
export const IonAvatar = createReactComponent('ion-avatar');
export const IonBackdrop = createReactComponent('ion-backdrop');
export const IonBadge = createReactComponent('ion-badge');
export const IonButton = createReactComponent('ion-button');
export const IonButtons = createReactComponent('ion-buttons');
export const IonCard = createReactComponent('ion-card');
export const IonCardContent = createReactComponent('ion-card-content');
export const IonCardHeader = createReactComponent('ion-card-header');
export const IonCardSubtitle = createReactComponent('ion-card-subtitle');
export const IonCardTitle = createReactComponent('ion-card-title');
export const IonCheckbox = createReactComponent('ion-checkbox');
export const IonCol = createReactComponent('ion-col');
export const IonContent = createReactComponent('ion-content');
export const IonChip = createReactComponent('ion-chip');
export const IonDatetime = createReactComponent('ion-datetime');
export const IonFab = createReactComponent('ion-fab');
export const IonFabButton = createReactComponent('ion-fab-button');
export const IonFabList = createReactComponent('ion-fab-list');
export const IonFooter = createReactComponent('ion-footer');
export const IonGrid = createReactComponent('ion-grid');
export const IonHeader = createReactComponent('ion-header');
export const IonImg = createReactComponent('ion-img');
export const IonInfiniteScroll = createReactComponent('ion-infinite-scroll');
export const IonInput = createReactComponent('ion-input');
export const IonItem = createReactComponent('ion-item');
export const IonItemDivider = createReactComponent('ion-item-divider');
export const IonItemGroup = createReactComponent('ion-item-group');
export const IonItemOption = createReactComponent('ion-item-option');
export const IonItemOptions = createReactComponent('ion-item-options');
export const IonItemSliding = createReactComponent('ion-item-sliding');
export const IonLabel = createReactComponent('ion-label');
export const IonList = createReactComponent('ion-list');
export const IonListHeader = createReactComponent('ion-list-header');
export const IonMenu = createReactComponent('ion-menu');
export const IonMenuButton = createReactComponent('ion-menu-button');
export const IonMenuToggle = createReactComponent('ion-menu-toggle');
export const IonNote = createReactComponent('ion-note');
export const IonPicker = createReactComponent('ion-picker');
export const IonPickerColumn = createReactComponent('ion-picker-column');
export const IonNav = createReactComponent('ion-nav');
export const IonProgressBar = createReactComponent('ion-progress-bar');
export const IonRadio = createReactComponent('ion-radio');
export const IonRadioGroup = createReactComponent('ion-radio-group');
export const IonRange = createReactComponent('ion-range');
export const IonRefresher = createReactComponent('ion-refresher');
export const IonRefresherContent = createReactComponent('ion-refresher-content');
export const IonReorder = createReactComponent('ion-reorder');
export const IonReorderGroup = createReactComponent('ion-reorder-group');
export const IonRippleEffect = createReactComponent('ion-ripple-effect');
export const IonRow = createReactComponent('ion-row');
export const IonSearchbar = createReactComponent('ion-searchbar');
export const IonSegment = createReactComponent('ion-segment');
export const IonSegmentButton = createReactComponent('ion-segment-button');
export const IonSelect = createReactComponent('ion-select');
export const IonSelectOption = createReactComponent('ion-select-option');
export const IonSelectPopover = createReactComponent('ion-select-popover');
export const IonSkeletonText = createReactComponent('ion-skeleton-text');
export const IonSlide = createReactComponent('ion-slide');
export const IonSlides = createReactComponent('ion-slides');
export const IonSpinner = createReactComponent('ion-spinner');
export const IonSplitPane = createReactComponent('ion-split-pane');
export const IonText = createReactComponent('ion-text');
export const IonTextarea = createReactComponent('ion-textarea');
export const IonThumbnail = createReactComponent('ion-thumbnail');
export const IonTitle = createReactComponent('ion-title');
export const IonToggle = createReactComponent('ion-toggle');
export const IonToolbar = createReactComponent('ion-toolbar');
export const IonVirtualScroll = createReactComponent('ion-virtual-scroll');
